### PR TITLE
Adds an Endpoints sample with multiple versions.

### PR DESCRIPTION
**Pull request by wmwong**
This is a copy of the existing getting-started sample with extensions to support multiple versions.

The difference in the API between v1 and v2 is the use of "message" vs "msg".

The code sample shows:
- Simple input layer that receives API requests from multiple major versions (v1/v2 servlets).
- Simple translation to a backend format (v1/v2 translators).
- Simple backend processing (uppercasing the message).
- Simple translation to the response format (v1/v2 translators).

No integration tests were written (similar to the other Endpoints samples) because this sample doesn't call any services. Instead, it is meant to be a sample that is deployed to GCP.

### CODE REVIEW

File: endpoints/multiple-versions/container-engine.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# Copyright 2015 Google Inc. All Rights Reserved.
```
Comment (lesv) [inline endpoints/multiple-versions/container-engine.yaml]: Doesn't need All Rights Reserved. (for samples)

File: endpoints/multiple-versions/openapi-v2.yaml [hunk 2]
```diff
@@ -0,0 +1,108 @@
+# [START swagger]
```
Comment (lesv) [inline endpoints/multiple-versions/openapi-v2.yaml]: Should have a license

File: endpoints/multiple-versions/openapi-v2.yaml [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+host: "echo-api.endpoints.YOUR-PROJECT-ID.cloud.goog"
+# [END swagger]
+basePath: "/v2"
+consumes:
+- "application/json"
+produces:
+- "application/json"
+schemes:
+- "https"
+paths:
```
Comment (lesv) [inline endpoints/multiple-versions/openapi-v2.yaml]: Some whitespace would make this a bit more readable.

File: endpoints/multiple-versions/openapi.yaml [hunk 4]
```diff
@@ -0,0 +1,108 @@
+# [START swagger]
```
Comment (lesv) [inline endpoints/multiple-versions/openapi.yaml]: License & vertical whitespace

File: endpoints/multiple-versions/pom.xml [hunk 5]
```diff
@@ -0,0 +1,83 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (lesv) [inline endpoints/multiple-versions/pom.xml]: License

File: endpoints/multiple-versions/pom.xml [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        <version>2.0.9.121.v20160815</version>
+      </plugin>
+      <plugin>
+        <groupId>com.google.cloud.tools</groupId>
+        <artifactId>appengine-maven-plugin</artifactId>
+        <version>${appengine.maven.plugin}</version>
+        <configuration>
+        </configuration>
+      </plugin>
+      <plugin>
```
Comment (lesv) [inline endpoints/multiple-versions/pom.xml]: Not actually required if you are on 3.3.9 or later

File: endpoints/multiple-versions/pom.xml [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+      <artifactId>google-collections</artifactId>
+      <version>1.0-rc2</version>
+    </dependency>
+  </dependencies>
+
+  <build>
+    <!-- for hot reload of the web application -->
+    <outputDirectory>${project.build.directory}/${project.build.finalName}/WEB-INF/classes</outputDirectory>
+    <plugins>
+      <plugin>  <!-- TEMPORARY -->
```
Comment (lesv) [inline endpoints/multiple-versions/pom.xml]: Shouldn't this be the newer plugin?

File: endpoints/multiple-versions/src/main/appengine/app.yaml [hunk 8]
```diff
@@ -0,0 +1,13 @@
+runtime: custom
```
Comment (lesv) [inline endpoints/multiple-versions/src/main/appengine/app.yaml]: License??

File: endpoints/multiple-versions/src/main/docker/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,5 @@
+FROM gcr.io/google_appengine/jetty9
```
Comment (lesv) [inline endpoints/multiple-versions/src/main/docker/Dockerfile]: License ??

File: endpoints/multiple-versions/src/main/java/com/example/endpoints/AuthInfoServlet.java [hunk 10]
```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright 2015 Google Inc. All Rights Reserved.
```
Comment (lesv) [inline endpoints/multiple-versions/src/main/java/com/example/endpoints/AuthInfoServlet.java]: Strike "All Rights Reserved."

File: endpoints/multiple-versions/src/main/java/com/example/endpoints/Echo.java [hunk 11]
```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright 2015 Google Inc. All Rights Reserved.
```
Comment (lesv) [inline endpoints/multiple-versions/src/main/java/com/example/endpoints/Echo.java]: Strike "All Rights Reserved."

File: endpoints/multiple-versions/src/main/java/com/example/endpoints/EchoV2Servlet.java [hunk 12]
```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright 2015 Google Inc. All Rights Reserved.
```
Comment (lesv) [inline endpoints/multiple-versions/src/main/java/com/example/endpoints/EchoV2Servlet.java]: Strike "All Rights Reserved."

File: endpoints/multiple-versions/src/main/java/com/example/endpoints/messsage/Message.java [hunk 13]
```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright 2015 Google Inc. All Rights Reserved.
```
Comment (lesv) [inline endpoints/multiple-versions/src/main/java/com/example/endpoints/messsage/Message.java]: Strike "All Rights Reserved."

File: endpoints/multiple-versions/src/main/java/com/example/endpoints/messsage/MessageTranslator.java [hunk 14]
```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright 2015 Google Inc. All Rights Reserved.
```
Comment (lesv) [inline endpoints/multiple-versions/src/main/java/com/example/endpoints/messsage/MessageTranslator.java]: Strike ARR.

File: endpoints/multiple-versions/src/main/java/com/example/endpoints/messsage/MessageV1Translator.java [hunk 15]
```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright 2015 Google Inc. All Rights Reserved.
```
Comment (lesv) [inline endpoints/multiple-versions/src/main/java/com/example/endpoints/messsage/MessageV1Translator.java]: Strike ARR

File: endpoints/multiple-versions/src/main/java/com/example/endpoints/messsage/MessageV2Translator.java [hunk 16]
```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright 2015 Google Inc. All Rights Reserved.
```
Comment (lesv) [inline endpoints/multiple-versions/src/main/java/com/example/endpoints/messsage/MessageV2Translator.java]: Strike ARR

File: endpoints/multiple-versions/container-engine.yaml (hunk 1 above)
Comment (wmwong) [inline endpoints/multiple-versions/container-engine.yaml]: Done

File: endpoints/multiple-versions/openapi-v2.yaml (hunk 2 above)
Comment (wmwong) [inline endpoints/multiple-versions/openapi-v2.yaml]: Done

File: endpoints/multiple-versions/openapi-v2.yaml (hunk 3 above)
Comment (wmwong) [inline endpoints/multiple-versions/openapi-v2.yaml]: Done

File: endpoints/multiple-versions/openapi.yaml (hunk 4 above)
Comment (wmwong) [inline endpoints/multiple-versions/openapi.yaml]: Done

File: endpoints/multiple-versions/openapi-v2.yaml (hunk 3 above)
Comment (wmwong) [inline endpoints/multiple-versions/openapi-v2.yaml]: I indented all the lists using https://github.com/OAI/OpenAPI-Specification/tree/master/examples/v2.0/yaml as examples. They seem to indent '-'. Also, if the item is more than one line, the '-' is on its own line.

File: endpoints/multiple-versions/pom.xml (hunk 5 above)
Comment (wmwong) [inline endpoints/multiple-versions/pom.xml]: Done

File: endpoints/multiple-versions/pom.xml (hunk 7 above)
Comment (wmwong) [inline endpoints/multiple-versions/pom.xml]: https://github.com/GoogleCloudPlatform/gcloud-maven-plugin

The doc uses 2.0.9.121.v20160815.

File: endpoints/multiple-versions/pom.xml (hunk 6 above)
Comment (wmwong) [inline endpoints/multiple-versions/pom.xml]: Removed.

File: endpoints/multiple-versions/src/main/appengine/app.yaml (hunk 8 above)
Comment (wmwong) [inline endpoints/multiple-versions/src/main/appengine/app.yaml]: Done

File: endpoints/multiple-versions/src/main/docker/Dockerfile (hunk 9 above)
Comment (wmwong) [inline endpoints/multiple-versions/src/main/docker/Dockerfile]: Done

File: endpoints/multiple-versions/src/main/java/com/example/endpoints/AuthInfoServlet.java (hunk 10 above)
Comment (wmwong) [inline endpoints/multiple-versions/src/main/java/com/example/endpoints/AuthInfoServlet.java]: Done

File: endpoints/multiple-versions/src/main/java/com/example/endpoints/Echo.java (hunk 11 above)
Comment (wmwong) [inline endpoints/multiple-versions/src/main/java/com/example/endpoints/Echo.java]: Done

File: endpoints/multiple-versions/src/main/java/com/example/endpoints/EchoV2Servlet.java (hunk 12 above)
Comment (wmwong) [inline endpoints/multiple-versions/src/main/java/com/example/endpoints/EchoV2Servlet.java]: Done

File: endpoints/multiple-versions/src/main/java/com/example/endpoints/messsage/Message.java (hunk 13 above)
Comment (wmwong) [inline endpoints/multiple-versions/src/main/java/com/example/endpoints/messsage/Message.java]: Done

File: endpoints/multiple-versions/src/main/java/com/example/endpoints/messsage/MessageTranslator.java (hunk 14 above)
Comment (wmwong) [inline endpoints/multiple-versions/src/main/java/com/example/endpoints/messsage/MessageTranslator.java]: Done

File: endpoints/multiple-versions/src/main/java/com/example/endpoints/messsage/MessageV1Translator.java (hunk 15 above)
Comment (wmwong) [inline endpoints/multiple-versions/src/main/java/com/example/endpoints/messsage/MessageV1Translator.java]: Done

File: endpoints/multiple-versions/src/main/java/com/example/endpoints/messsage/MessageV2Translator.java (hunk 16 above)
Comment (wmwong) [inline endpoints/multiple-versions/src/main/java/com/example/endpoints/messsage/MessageV2Translator.java]: Done

File: endpoints/multiple-versions/pom.xml (hunk 7 above)
Comment (lesv) [inline endpoints/multiple-versions/pom.xml]: I meant the [other one](https://cloud.google.com/appengine/docs/flexible/java/using-maven#adding_the_app_engine_maven_and_jetty_maven_plugins):
```
<plugins>
   <plugin>
     <groupId>com.google.cloud.tools</groupId>
     <artifactId>appengine-maven-plugin</artifactId>
     <version>1.3.0</version>
   </plugin>
   <plugin>
     <groupId>org.eclipse.jetty</groupId>
     <artifactId>jetty-maven-plugin</artifactId>
     <version>9.4.3.v20170317</version>
   </plugin>
</plugins>
```

**Comment by wmwong** [comment]
Removed the old maven plugin. Tested locally and deployed to AppEngine. All working.

**Comment by lesv** [comment]
Argh - I thought I added the DO NOT MERGE tag -- sorry.

**Comment by wmwong** [comment]
I had added the DO NOT MERGE tag at the beginning. Looks like you might have accidentally removed it.

Should be fine I guess.

**Pull request merged**